text/0000-pack-buildpack-new-alternatives.md

```diff
@@ -0,0 +1,206 @@
+
+We will implement the project scaffolding logic in a Go module separate from `pack`.
+
+# Definitions
+[definitions]: #definitions
+
+* **project scaffolding**: minimal code and file system structure used to start a project.
+
+# Motivation
+[motivation]: #motivation
+
+The creation of buildpacks in languages other than `bash` is undocumented in the [Buildpack Author Guide](https://buildpacks.io/docs/buildpack-author-guide).  In particular, creating a buildpack in Go using `libcnb` is undocumented because there is no mechanism to generate a scaffolded project.  This proposal adds a minimal Go/`libcnb` project template to `pack` and allows 3d parties to provide templates to create buildpacks for their chosen technology stack.
+
+`pack buildpack create` supports end-users who wish to scaffold a new bash or Go/`libcnb`-based buildpack.  The operation of `pack buildpack create` requires Internet access to succeed.  `pack buildpack create --template URL` allows projects (such as Paketo, Heroku and others) to define skeleton projects for new buildpacks.
+
+Replacing `pack buildpack new` with `pack buildpack create` allows buildpack authors to more-easily create new buildpacks in their chosen language and framework.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Project scaffolding is [very](https://cookiecutter.readthedocs.io/) [popular](https://yeoman.io/) [in](https://github.com/kowainik/summoner) [other](https://crates.io/crates/cargo-scaffold) [ecosystems](https://github.com/golang-standards/project-layout).  Scaffolding systems are employed to ease onboarding of new developers.  Within `pack` this feature is targeted at onboarding buildpack authors.
+
+`pack buildpack create` should prompt the end user to choose an implementation language.  Thereafter, generate skeleton code according to the best practices for that language.  For example, users choosing Go as an implementation language would generate a [golang standard project layout](https://github.com/golang-standards/project-layout) and include `libcnb` best practices.  The layout would look similar to the following:
+
+```bash
+.
+├── buildpack.toml
+├── cmd
+│   └── main.go
+├── go.mod
+├── go.sum
+└── pkg
+    ├── build.go
+    └── detect.go
+```
+
+This approach is not opinionated on topics such as which testing framework to use.  Only components under github.com/buildpacks/* are used in the generated project.  `pack buildpack create --template URL` allows for more opinionated project scaffolding.  As such, we support new buildpack authors with onboarding and support experienced buildpack authors to use the scaffolding of their choice.
```

Comment:
Does the `--template` flag support subdirectories? I noticed [this repository you mentioned](https://github.com/AidanDelaney/cnb-buildpack-templates) has multiple templates. Would this repository be supported or would you need to create a repository per template?

Reply:
The `--template` flag supports both templates and template collections.  A template collection is a repository that contains templates in subdirectories.